charts/spinnaker/templates/configmap/halyard-config.yaml

```diff
@@ -109,49 +109,29 @@ data:
     {{- end }}
 
     $HAL_COMMAND config provider kubernetes enable
-    {{- if .Values.kubeConfig.accountPerContext }}
-    {{- range $index, $context := .Values.kubeConfig.contexts }}
+    {{- range $accountName, $values := .Values.kubeConfig.accounts }}
 
-    if $HAL_COMMAND config provider kubernetes account get {{ $context }}; then
+    if $HAL_COMMAND config provider kubernetes account get {{ $accountName }}; then
       PROVIDER_COMMAND='edit'
     else
       PROVIDER_COMMAND='add'
     fi
 
-    $HAL_COMMAND config provider kubernetes account $PROVIDER_COMMAND {{ $context }} --docker-registries dockerhub \
-                --context {{ $context }} {{ if not $.Values.kubeConfig.enabled }}--service-account true{{ end }} \
+    $HAL_COMMAND config provider kubernetes account $PROVIDER_COMMAND {{ $accountName }} --docker-registries dockerhub \
+                --context {{ $values.context }} {{ if not $.Values.kubeConfig.enabled }}--service-account true{{ end }} \
                 {{ if $.Values.kubeConfig.enabled }}--kubeconfig-file {{ template "spinnaker.kubeconfig" $ }}{{ end }} \
                 {{ if $.Values.kubeConfig.onlySpinnakerManaged.enabled }}--only-spinnaker-managed true{{ end }} \
                 {{ if not $.Values.kubeConfig.checkPermissionsOnStartup }}--check-permissions-on-startup false{{ end }} \
-                {{ if $.Values.kubeConfig.nameSpaces }}--namespaces={{ template "nameSpaces" $ }}{{ end }} \
-                {{ if not $.Values.kubeConfig.nameSpaces }}--omit-namespaces={{ template "omittedNameSpaces" $ }}{{ end }} \
+                {{ if $values.namespaces }}--namespaces={{ join "," $values.namespaces }}{{ end }} \
+                {{ if not $values.namespaces }}--omit-namespaces={{ template "omittedNameSpaces" $ }}{{ end }} \
```

Comment:
Should this not prefer an `omittedNamespaces` on `$values`, and only if that is not set, apply the global setting?

Reply:
So your question is whether `omittedNamespaces` should be global, account-bound or have the possibility of both right? 
The latter indeed makes sense to me. Will try out something tomorrow to conditionally take account or global `omittedName[s|S]paces`.

Reply:
Pushed something, PTAL :)